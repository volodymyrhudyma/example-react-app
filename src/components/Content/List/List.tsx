import React, { Component } from 'react';
import {
  Wrapper,
  LineWrapper,
  ListHeading,
  ListBody,
} from './styled/List.styled';
import Title from 'src/ui-kit/Title';
import Line from 'src/ui-kit/Line';
import Article from 'src/components/Content/ListItem';
import { CARD } from 'src/constants/layout';
import { IArticle } from 'src/types';
import Loader from 'src/ui-kit/Loader';
import Error from 'src/ui-kit/Error';
import Empty from 'src/components/Empty';
import { WithTranslation } from 'react-i18next';

interface IListProps extends WithTranslation {
  layout: string;
  fetchArticles: () => void;
  fetching: boolean;
  error?: string;
  data: IArticle[];
}

class List extends Component<IListProps> {
  componentDidMount() {
    const { fetchArticles } = this.props;
    fetchArticles();
  }
  render() {
    const { t, layout, fetching, error, data } = this.props;
    return (
      <Wrapper>
        <ListHeading>
          <Title type="h5">{t('articles.period')}</Title>
          <LineWrapper>
            <Line width="50px" />
          </LineWrapper>
        </ListHeading>
        <ListBody flex={layout === CARD}>
          {fetching ? (
            <Loader />
          ) : error ? (
            <Error message={t('resource.error')} />
          ) : data.length > 0 ? (
            data.map(article => <Article key={article.id} article={article} />)
          ) : (
            <Empty />
          )}
        </ListBody>
      </Wrapper>
    );
  }
}

export default List;
