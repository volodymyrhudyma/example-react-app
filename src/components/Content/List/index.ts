import { connect } from 'react-redux';
import { IAppState } from 'src/types';
import Component from './List';
import { fetchArticles } from 'src/actions/article';
import { withTranslation } from 'react-i18next';

const mapStateToProps = (state: IAppState) => ({
  layout: state.layout.layout,
  fetching: state.article.articles.fetching,
  error: state.article.articles.error,
  data: state.article.articles.data,
});

const mapDispatchToProps = (dispatch: any) => {
  return {
    fetchArticles: () => dispatch(fetchArticles()),
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withTranslation()(Component));
