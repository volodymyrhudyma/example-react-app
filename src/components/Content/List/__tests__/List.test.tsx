import { shallow } from 'enzyme';
import React from 'react';
import List from '../List';
import i18n from 'i18next';
import Article from '../../../Content/ListItem';

describe('List with Enzyme and Jest', () => {
  let wrapper: any = null;
  const fetchArticles = jest.fn();
  const articles = [
    {
      avatar: '',
      author: '',
      email: '',
      isRead: false,
      jobTitle: '',
      title: '',
      desc: '',
      tags: [],
      id: 0,
    },
    {
      avatar: '',
      author: '',
      email: '',
      isRead: false,
      jobTitle: '',
      title: '',
      desc: '',
      tags: [],
      id: 0,
    },
  ];
  beforeEach(() => {
    wrapper = shallow(
      <List
        t={jest.fn()}
        i18n={i18n}
        tReady={false}
        layout="TABLE"
        fetchArticles={fetchArticles}
        fetching={false}
        error={undefined}
        data={articles}
      />
    );
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should fetch all articles', () => {
    expect(fetchArticles).toHaveBeenCalled();
  });
  it('should show 2 Article components if 2 articles are returned by API', () => {
    expect(wrapper.find(Article).length).toBe(2);
  });
});
