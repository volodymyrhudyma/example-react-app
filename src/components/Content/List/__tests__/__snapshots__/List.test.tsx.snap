// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List with Enzyme and Jest should render correctly 1`] = `
<styled.div>
  <styled.div>
    <Title
      type="h5"
    />
    <styled.div>
      <Line
        height="4px"
        width="50px"
      />
    </styled.div>
  </styled.div>
  <styled.div
    flex={false}
  >
    <Connect(withI18nextTranslation(ListItem))
      article={
        Object {
          "author": "",
          "avatar": "",
          "desc": "",
          "email": "",
          "id": 0,
          "isRead": false,
          "jobTitle": "",
          "tags": Array [],
          "title": "",
        }
      }
      key="0"
    />
    <Connect(withI18nextTranslation(ListItem))
      article={
        Object {
          "author": "",
          "avatar": "",
          "desc": "",
          "email": "",
          "id": 0,
          "isRead": false,
          "jobTitle": "",
          "tags": Array [],
          "title": "",
        }
      }
      key="0"
    />
  </styled.div>
</styled.div>
`;
